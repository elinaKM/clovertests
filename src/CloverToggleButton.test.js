import React from 'react';
import renderer from 'react-test-renderer';

import {CloverToggleButton} from 'cloveruicomponents';

describe('CloverToggleButton', () => {

  var handler = () => {
    console.log("I was pressed!");
  }
  const secondary = "#272121";
  const primary = "yellow";
  const bgText = "#f6e9e9";
  const positive = "#BAEDBF";
  const negative = "#FCA6A6";

  var myFixedTheme = {
  palette: {
    body: "#363333",
    text: bgText,
    primary,
    primaryText: bgText,
    secondary,
    secondaryText: secondary,
    inverse: bgText,
    inverseText: "#363333",
    positive,
    negative,
    dark: "rgb(54, 51, 51)"
  }
}

  it('renders correctly with all properties set', () => {
    const tree = renderer
      .create(
        <CloverToggleButton 
          clickHandler={handler} 
          sizeOfFont="40px" 
          useTheme={myFixedTheme}>Roll
        </CloverToggleButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CloverToggleButton', () => {
  it('renders correctly with no properties set', () => {
    const tree = renderer
      .create(<CloverToggleButton>Roll</CloverToggleButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CloverToggleButton', () => {
  it('renders correctly with no title given', () => {
    const tree = renderer
      .create(<CloverToggleButton></CloverToggleButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});