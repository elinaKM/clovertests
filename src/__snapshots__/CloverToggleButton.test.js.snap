// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloverToggleButton renders correctly with all properties set 1`] = `
<button
  className="sc-bwzfXH jUAIJN"
  fontSize="40px"
  onClick={[Function]}
>
  Roll
</button>
`;

exports[`CloverToggleButton renders correctly with no properties set 1`] = `
<button
  className="sc-bwzfXH hxoqPB"
  fontSize={undefined}
  onClick={undefined}
>
  Roll
</button>
`;

exports[`CloverToggleButton renders correctly with no title given 1`] = `
<button
  className="sc-bwzfXH hxoqPB"
  fontSize={undefined}
  onClick={undefined}
/>
`;
